main.tf

```diff
@@ -38,47 +50,96 @@ resource "aws_efs_access_point" "default" {
+    for_each = local.secondary_gids[each.key] != null ? ["true"] : []
+
+    content {
+      gid            = var.access_points[each.key]["posix_user"]["gid"]
+      uid            = var.access_points[each.key]["posix_user"]["uid"]
+      secondary_gids = local.secondary_gids[each.key] != null ? split(",", local.secondary_gids[each.key]) : null
+    }
   }
 
   root_directory {
     path = "/${each.key}"
-    creation_info {
-      owner_gid   = var.access_points[each.key]["creation_info"]["gid"]
-      owner_uid   = var.access_points[each.key]["creation_info"]["uid"]
-      permissions = var.access_points[each.key]["creation_info"]["permissions"]
+
+    dynamic "creation_info" {
+      for_each = try(lookup(var.access_points[each.key]["creation_info"]["gid"], ""), "") != "" ? ["true"] : []
+
+      content {
+        owner_gid   = var.access_points[each.key]["creation_info"]["gid"]
+        owner_uid   = var.access_points[each.key]["creation_info"]["uid"]
+        permissions = var.access_points[each.key]["creation_info"]["permissions"]
+      }
     }
   }
 
   tags = module.this.tags
 }
 
+module "security_group_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+
+  # For backwards compatibility with v0.30.1
+  attributes = concat(module.this.attributes, [var.security_group_suffix])
+
+  context = module.this.context
+}
+
 module "security_group" {
   source  = "cloudposse/security-group/aws"
-  version = "0.3.1"
+  version = "0.4.2"
 
-  use_name_prefix = var.security_group_use_name_prefix
-  rules           = var.security_group_rules
-  vpc_id          = var.vpc_id
-  description     = var.security_group_description
+  enabled                       = local.security_group_enabled
+  security_group_name           = var.security_group_name
+  create_before_destroy         = var.security_group_create_before_destroy
+  security_group_create_timeout = var.security_group_create_timeout
+  security_group_delete_timeout = var.security_group_delete_timeout
 
-  enabled = local.security_group_enabled
-  context = module.this.context
+  security_group_description = var.security_group_description
+  allow_all_egress           = true
+  rules                      = var.additional_security_group_rules
+  rule_matrix = [
+    {
+      source_security_group_ids = local.allowed_security_group_ids
+      cidr_blocks               = var.allowed_cidr_blocks
+      rules = [
+        {
+          key         = null
+          type        = "ingress"
+          from_port   = 2049
+          to_port     = 2049
+          protocol    = "tcp"
+          description = "Allow ingress EFS traffic"
+        }
+      ]
+    }
+  ]
+  vpc_id = var.vpc_id
+
+  context = module.security_group_label.context
 }
 
 module "dns" {
   source  = "cloudposse/route53-cluster-hostname/aws"
-  version = "0.12.0"
+  version = "0.12.2"
 
-  enabled  = module.this.enabled && length(var.zone_id) > 0 ? true : false
+  enabled  = local.enabled && length(var.zone_id) > 0
```

Comment:
For this to work when the DNS zone is being created in the same Terraform plan, `zone_id` has to be redefined. See https://github.com/cloudposse/terraform-aws-elasticache-redis/issues/82
```hcl
variable "zone_id" {
  type        = any
  default     = []
  description = <<-EOT
    Route53 DNS Zone ID as list of string (0 or 1 items). If empty, no custom DNS name will be published.
    If the list contains a single Zone ID, a custom DNS name will be pulished in that zone.
    Can also be a plain string, but that use is DEPRECATED because of Terraform issues.
    EOT
}
```